components/cloud_subscribe_with_loading_modal/index.tsx

```diff
@@ -0,0 +1,246 @@
+            setTimeout(
+                completeSubscribe,
+                MIN_PROCESSING_MILLISECONDS - millisecondsElapsed,
+            );
+            return;
+        }
+        completeSubscribe();
+    };
+
+    const completeSubscribe = () => {
+        clearInterval(intervalId);
+        setProcessingState(ProcessState.SUCCESS);
+        progress.current = 100;
+    };
+
+    const updateProgress = () => {
+        if (progress.current >= MAX_FAKE_PROGRESS) {
+            clearInterval(intervalId);
+            return;
+        }
+        progress.current =
+            progress.current + 3 > MAX_FAKE_PROGRESS ? MAX_FAKE_PROGRESS : progress.current + 3;
+    };
+
+    const handleGoBack = () => {
+        clearInterval(intervalId);
+        progress.current = 0;
+        setError(false);
+        setProcessingState(ProcessState.PROCESSING);
+        props.onBack();
+    };
+
+    const handleClose = () => {
+        dispatch(
+            closeModal(
+                ModalIdentifiers.CLOUD_SUBSCRIBE_WITH_LOADING_MODAL,
+            ),
+        );
+        props.onClose();
+    };
+
+    const sucessPage = () => {
+        const formattedBtnText = (
+            <FormattedMessage
+                defaultMessage='Return to {team}'
+                id='admin.billing.subscription.returnToTeam'
+                values={{team: props.teamToKeep?.display_name}}
+            />
+        );
+        const productName = props.selectedProduct?.name;
+        const title = (
+            <FormattedMessage
+                id={'admin.billing.subscription.downgradedSuccess'}
+                defaultMessage={"You're now subscribed to {productName}"}
+                values={{productName}}
+            />
+        );
+
+        const formattedSubtitle = (
+            <FormattedMessage
+                id='success_modal.subtitle'
+                defaultMessage='Your final bill will be prorated. Your workspace now has {productName} limits'
+                values={{productName}}
+            />
+        );
+        return (
+            <IconMessage
+                formattedTitle={title}
+                formattedSubtitle={formattedSubtitle}
+                error={error}
+                icon={
+                    <PaymentSuccessStandardSvg
+                        width={444}
+                        height={313}
+                    />
+                }
+                formattedButtonText={formattedBtnText}
+                buttonHandler={handleClose}
+                className={'success'}
+                tertiaryBtnText={t('admin.billing.subscription.viewBilling')}
+                tertiaryButtonHandler={() => {
+                    handleClose();
+                    props.history.push(
+                        '/admin_console/billing/subscription',
+                    );
+                }}
+            />
+        );
+    };
+
+    const modalContent = () => {
+        const progressBar: JSX.Element | null = (
```

Comment:
Could this be a separated component, so that can be reused across the webapp?
Maybe also the progress calculations can be taken out for reusability.

Reply:
Split it out to a separate component.